cpp/src/arrow/type.h

```diff
@@ -456,7 +456,7 @@ struct ARROW_EXPORT Date32Type : public DateType {
 
   Date32Type();
 
-  int bit_width() const override { return static_cast<int>(sizeof(c_type) * 4); }
+  int bit_width() const override { return static_cast<int>(sizeof(c_type) * 8); }
```

Comment:
there's `CHAR_BIT`, which is the number of bits in a byte. We should probably use that here.